### PR TITLE
Wait for app ready to delete old Semeru instances

### LINKED ISSUE #345: Old Semeru instances are cleaned up too quickly
Noticed that the old Semeru compiler instances are cleaned up quickly before the old application Pods go away. It's more apparent when the application upgrade takes long(er) time, which would be the case with this combination of update strategy and higher initial delay for readiness probe.

I believe that checking for reconcile completion is not sufficient. We also need to check that the status of `Ready` condition is true and the message is `Application is reconciled and resources are ready.` 

```
apiVersion: liberty.websphere.ibm.com/v1
kind: WebSphereLibertyApplication
metadata:
  name: sample-app
  namespace: test
spec:
  license:
    accept: true
    edition: IBM WebSphere Application Server
    productEntitlementSource: Standalone
  probes:
    readiness:
      initialDelaySeconds: 30
  deployment:
    updateStrategy:
      rollingUpdate:
        maxSurge: 1
        maxUnavailable: 1
  applicationImage: 'icr.io/appcafe/open-liberty/samples/getting-started:latest'
  manageTLS: true
  replicas: 6
```

### PR DESCRIPTION
**Pull request by kabicin**
Fixes #345.
- Waits until pod updates by checking application `Ready` state as true and with message `common.StatusConditionTypeReadyMessage` which is a variable created in https://github.com/application-stacks/runtime-component-operator/pull/438

### CODE REVIEW

**Review by leochr — APPROVED** [review]
@kabicin Looks good. Thank you

File: controllers/webspherelibertyapplication_controller.go [hunk 1]
```diff
@@ -670,6 +672,14 @@ func (r *ReconcileWebSphereLiberty) Reconcile(ctx context.Context, request ctrl.
 	return r.ManageSuccess(common.StatusConditionTypeReconciled, instance)
 }
 
+func (r *ReconcileWebSphereLiberty) isWebSphereLibertyApplicationReady(ba common.BaseComponent) bool {
+	if r.CheckApplicationStatus(ba) == corev1.ConditionTrue {
```
Comment (leochr) [inline controllers/webspherelibertyapplication_controller.go:676]: Wondering if the call to `CheckApplicationStatus` can be omitted as it's checked in ManageSuccess. The removal of old semeru instance may only happen in the next reconcile, which would probably be okay.

File: controllers/webspherelibertyapplication_controller.go [hunk 2]
```diff
@@ -657,11 +657,13 @@ func (r *ReconcileWebSphereLiberty) Reconcile(ctx context.Context, request ctrl.
 		reqLogger.V(1).Info(fmt.Sprintf("%s is not supported", prometheusv1.SchemeGroupVersion.String()))
 	}
 
-	// Delete completed Semeru instances because all pods now point to the newest Semeru service
-	if areCompletedSemeruInstancesMarkedToBeDeleted {
-		if err := r.deleteCompletedSemeruInstances(instance); err != nil {
-			reqLogger.Error(err, "Failed to delete completed Semeru instance")
-			return r.ManageError(err, common.StatusConditionTypeReconciled, instance)
+	if r.isWebSphereLibertyApplicationReady(instance) {
```
Comment (leochr) [inline controllers/webspherelibertyapplication_controller.go:660]: It'll be good to do this check after checking areCompletedSemeruInstancesMarkedToBeDeleted, so status is not checked unnecessarily.

**Review by leochr — COMMENTED** [review]
@kabicin Added couple of comments. Please take a look when you can. Thanks.

Unrelated to this PR: It'll be good to move the currentGeneration under the if-check to avoid checking it when semeru is not enabled.

File: controllers/webspherelibertyapplication_controller.go (hunk 2 above)
Comment (kabicin) [reply controllers/webspherelibertyapplication_controller.go]: Done

**Review by leochr — APPROVED** [review]

**Pull request merged**